fakeredis/_server.py

```diff
@@ -664,7 +666,12 @@ def __init__(self, server):
         self._parser.send(None)
 
     def put_response(self, msg):
-        self.responses.put(msg)
+        # redis.Connection.__del__ might call self.close at any time, which
+        # will set self.responses to None. We assume this will happen
+        # atomically, and the code below then protects us against this.
+        responses = self.responses
+        if responses:
+            self.responses.put(msg)
```

Comment:
Should this not be `responses.put(msg)`? Otherwise it could still hit `None` the second time around.

Reply:
Doh! Yes, that's what I had in my head.

Reply:
Fixed, thanks.